x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/MlMemoryTracker.java

```diff
@@ -250,15 +250,11 @@ public Long getJobMemoryRequirement(String taskName, String id) {
             return null;
         }
 
-        if (MlTasks.TRAINED_MODEL_DEPLOYMENT_TASK_NAME.equals(taskName)) {
-            return getTrainedModelTaskMemoryRequirement(id);
-        } else {
-            Map<String, Long> memoryRequirementByJob = memoryRequirementByTaskName.get(taskName);
-            if (memoryRequirementByJob == null) {
-                return null;
-            }
-            return memoryRequirementByJob.get(id);
+        Map<String, Long> memoryRequirementByJob = memoryRequirementByTaskName.get(taskName);
+        if (memoryRequirementByJob == null) {
+            return null;
```

Comment:
I think it would be worth adding an `assert false` before this line, so that anyone who accidentally calls this method with a task name that isn't even in the outer map gets an in their test that leads to the root cause rather than having to figure it out from unexpected assignment of tasks to nodes.